chia/cmds/keys.py

```diff
@@ -78,9 +80,9 @@ def delete_cmd(ctx: click.Context, fingerprint: int):
 
 @keys_cmd.command("delete_all", short_help="Delete all private keys in keychain")
 def delete_all_cmd():
-    from .keys_funcs import keychain
+    from chia.util.keychain import Keychain
 
-    keychain.delete_all_keys()
+    Keychain().delete_all_keys()
```

Comment:
Why is `Keychain()` any better than `keychain`? They both have implicit global state associated with them. What motivated this change?

Reply:
Truthfully I'm not super happy with this, but it was done to support the tests that were written to cover the `keys` commands.

You're right that the `Keychain` wraps some global state held by `KeyringWrapper`. The individual tests will construct a temporary backing keyring, which gets stuffed into the global KeyringWrapper. This presented an issue with the existing `keyring` global instance which held a separate KeyringWrapper instance that wasn't configured properly for testing. The telltale symptom was that when running the tests, the "(Unlock Keyring) Passphrase:" prompt would appear, blocking the tests. So really this was done to ensure that the tests could setup the KeyringWrapper *before* keys.py created its global keychain.

Outside of the test environments, creating a global keychain or many Keychain() instances is effectively the same, minus the cost to construct the objects (which is pretty lightweight at this point).